R/unaids-navigator.R

```diff
@@ -53,16 +58,111 @@ write_navigator_checklist <- function(naomi_output,
              t_("NAVIGATOR_CAL_ART_DESC"),
              t_("NAVIGATOR_CAL_KOS_DESC"),
              t_("NAVIGATOR_CAL_NEW_INFECTIONS_DESC"),
-             t_("NAVIGATOR_CAL_METHOD_DESC"))                        
+             t_("NAVIGATOR_CAL_METHOD_DESC"))
 
   v <- data.frame(NaomiCheckPermPrimKey = key,
                   NaomiCheckDes = label,
-                  TrueFalse = NA)
+                  TrueFalse = FALSE)
 
+  ## Do checks
   ## These checks are always 'TRUE'
   always_true_checks <- c("Package_created", "Package_has_all_data", "Model_fit")
   v$TrueFalse[v$NaomiCheckPermPrimKey %in% always_true_checks] <- TRUE
 
+  # TODO: Compare aggregated naomi inputs to national Spectrum totals
+  # This is dummy code
+
+
+  ## Check for correct model options selection
+  valid_opt <- yaml::read_yaml(system.file("metadata/navigator_validation.yml", package = "naomi"))
+
+  # Is most recent calendar quarter selected
+  if(options$calendar_quarter_t2 == valid_opt$calendar_quarter_t2){
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_recent_qtr"] <- TRUE}
+
+  # Is future projection quarter selected
+  if(options$calendar_quarter_t2 == valid_opt$calendar_quarter_t2){
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_future_proj_qtr"] <- TRUE}
+
+  # Is ART attendance selected
+  if(options$artattend){v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_ART_attendance_yes"] <- TRUE}
```

Comment:
A bit of an aside, but I prefer this one:
```
v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_ART_attendance_yes"] <- options$artattend
```
over this one:
```
v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_ART_attendance_yes"] <- options$artattend
```

Curious why your preference for the latter?  (The latter is a bit safer, because it does not risk auto-completion of list elements; in particular it is much safer to write `options$artattend` as `options[["artattend"]]`, since there is another list element `options$artattend_t2`.  I remember exactly where I was sitting--on my sister's sofa--when I debugged that over Christmas 2019.)